### ISSUE TITLE
get rid of == override for KeyDetails

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#1547]
There are three ways to possibly compare key details:
 - is the fingerprint the same?
 - is the fingerprint and last modified the same?
 - is all of the information the same?

Whenever one uses `==` I'd rather that the comparison was explicit like `keyDetails1.id == keyDetails2.id` or similar.

Therefore I'd like to remove 

```swift
    static func == (lhs: KeyDetails, rhs: KeyDetails) -> Bool {
        lhs.ids == rhs.ids
    }
```

To prevent unnecessary surprises.

**Comment by tomholub** [comment]
also 

```switf
    // (tom) todo - this is a confusing thing to do
    // when comparing pass phrases to one another, you would expect that it's compared by the pass phrase string
    // itself, and not by primary fingerprint of the associated key. I understand this is being used somewhere,
    // but I suggest to refactor it to avoid defining this == overload.
    static func == (lhs: PassPhrase, rhs: PassPhrase) -> Bool {
        lhs.primaryFingerprintOfAssociatedKey == rhs.primaryFingerprintOfAssociatedKey
    }
```

**Comment by tomholub** [comment]
This is a proposal, open to discuss.

**Comment by ioanmo226** [comment]
I agree with you. == might make confusion if we compare by KeyDetails.ids or fingerprints or something else.

**Comment by ioanmo226** [comment]
Here's my suggestion.

For PassPhrase, we can remove == operator.
For KeyDetails, we can't remove == operator as KeyDetails inherits from Decodable protocol and therefore it requires KeyDetails to inherit Equatable protocol.
So I suggest to keep == operator for KeyDetails but compare both fingerprint and lastModified.

Let me know your thoughts @tomholub, @sosnovsky

**Comment by tomholub** [comment]
You could have it compare all data in `KeyDetails`. Which should be the default for comparing structs - is there a way to do that without having to manually implement comparison of each field?

**Comment by ioanmo226** [comment]
Indeed there's a way, In KeyDetails, there was only one param[algo] (which didn't confirm to Equatable).
So I added equatable to KeyAlgo.
Please check PR.
